### PR TITLE
fix problems with restarting negentropy

### PR DESCRIPTION
**Pull request by savelyev-an**

### CODE REVIEW

File: vault-plugins/flant_iam/backend/factory.go [hunk 1]
```diff
@@ -108,7 +108,7 @@ func newBackend(conf *logical.BackendConfig) (logical.Backend, error) {
 
 	logger := conf.Logger.Named("IAM")
 
-	storage.AddKafkaSource(kafka_source.NewSelfKafkaSource(mb, restoreHandlers))
+	storage.AddKafkaSource(kafka_source.NewSelfKafkaSource(mb, restoreHandlers, conf.Logger.Named("KafkaSourceSelf")))
```
Comment (name212) [inline vault-plugins/flant_iam/backend/factory.go:111]: Maybe we will use `logger.Named("KafkaSourceSelf")`

Then, in logs, we will see `IAM.KafkaSourceSelf - some log`.

Iam auth has KafkaSourceSelf too

File: vault-plugins/shared/kafka/broker.go [hunk 2]
```diff
@@ -229,61 +235,60 @@ func (mb *MessageBroker) GetKafkaTransactionalProducer() *kafka.Producer {
...
+	defer mb.logger.Debug("SendMessages - exit")
+	if sourceInput != nil && (sourceInput.IgnoreBody || len(msgs) == 0) {
 		return mb.sendOffset(sourceInput)
 	}
 
 	if len(msgs) == 0 {
 		return nil
 	}
 
-	if len(msgs) == 1 && sourceInput == nil { // only single message without source
```
Comment (name212) [inline vault-plugins/shared/kafka/broker.go:283]: Why we delete this?

**Review by name212 — APPROVED** [review]

File: vault-plugins/flant_iam/backend/factory.go [hunk 3]
```diff
@@ -216,6 +216,15 @@ func newBackend(conf *logical.BackendConfig) (logical.Backend, error) {
 	)
 	localLogger.Debug("normal finish")
 
+	b.Clean = func(context.Context) {
+		l := b.Logger().Named("cleanup")
```
[the remark this answers is not in the document]
Comment (savelyev-an) [reply vault-plugins/flant_iam/backend/factory.go:220]: naming is composite (example from actual log):
2021-10-07T09:57:30.607Z [INFO]  auth.flant_iam_auth.auth_flant_iam_auth_29275f73.cleanup: started

File: vault-plugins/flant_iam_auth/backend/backend.go [hunk 4]
```diff
@@ -292,14 +292,18 @@ func (b *flantIamAuthBackend) SetupBackend(ctx context.Context, config *logical.
 }
 
 func (b *flantIamAuthBackend) cleanup(_ context.Context) {
+	l := b.Logger().Named("cleanup")
```
[the remark this answers is not in the document]
Comment (savelyev-an) [reply vault-plugins/flant_iam_auth/backend/backend.go:295]: see above

File: vault-plugins/shared/kafka/broker.go (hunk 2 above)
Comment (savelyev-an) [reply vault-plugins/shared/kafka/broker.go:283]: We have explored with Yury - using at the same time transactional commit reading and none-transactional leads to ambiguity in message offsets. So, we decide to use only transactional commit read.

**Pull request merged**